l10n_es_aeat_sii/models/account_invoice.py

```diff
@@ -127,6 +145,26 @@ def onchange_fiscal_position_l10n_es_aeat_sii(self):
                 key = invoice.fiscal_position.sii_registration_key_purchase
             invoice.sii_registration_key = key
 
+    @api.onchange('invoice_line')
+    def _onchange_invoice_line_l10n_es_aeat_sii(self):
+        for invoice in self:
+            company = invoice.company_id
+            method_desc = company.sii_description_method
+            header_sale = company.sii_header_sale
+            header_purchase = company.sii_header_purchase
+            description = ''
+            if invoice.type in ['out_invoice', 'out_refund'] and header_sale:
+                description = header_sale
+            elif invoice.type in ['in_invoice', 'in_refund'] and \
+                    header_purchase:
+                description = header_purchase
+            if method_desc == 'auto':
+                if len(description) > 0:
+                    description += ' | '
+                for line in invoice.invoice_line:
```

Comment:
Seria mejor hacer  ' - '.join([x.name for x in invoice.invoice_line]) para evitar el guión que se va a quedar al final